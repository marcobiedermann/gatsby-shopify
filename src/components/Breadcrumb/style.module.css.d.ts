declare const styles: {
  readonly breadcrumb: string;
  readonly breadcrumb__item: string;
};
export = styles;
