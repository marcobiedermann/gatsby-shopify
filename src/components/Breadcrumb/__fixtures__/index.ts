import { BreadcrumbProps } from '..';

const breadcrumb: BreadcrumbProps = {
  breadcrumbItems: [
    {
      name: 'Home',
      path: '/',
    },
  ],
};

export default breadcrumb;
