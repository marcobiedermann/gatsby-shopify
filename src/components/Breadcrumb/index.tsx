import { Link } from 'gatsby';
import React from 'react';
import { breadcrumb, breadcrumb__item } from './style.module.css';

export interface BreadcrumbItem {
  name: string;
  path: string;
}

export interface BreadcrumbProps {
  breadcrumbItems: BreadcrumbItem[];
}

function Breadcrumb(props: BreadcrumbProps) {
  const { breadcrumbItems } = props;

  return (
    <ol className={breadcrumb}>
      {breadcrumbItems.map((breadcrumbItem) => (
        <li className={breadcrumb__item} key={breadcrumbItem.path}>
          <Link to={breadcrumbItem.path}>{breadcrumbItem.name}</Link>
        </li>
      ))}
    </ol>
  );
}

export default Breadcrumb;
