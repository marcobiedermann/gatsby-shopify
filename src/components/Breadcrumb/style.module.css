.breadcrumb {
  display: flex;
  gap: 0.5em;
  list-style: none;
  margin-bottom: 1.5em;
  margin-top: 0;
  padding: 0;
}

.breadcrumb__item:not(:last-child)::after {
  content: 'Â»';
  margin-left: 0.5em;
}
