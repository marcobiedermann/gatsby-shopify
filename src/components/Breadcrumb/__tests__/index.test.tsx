import { render } from '@testing-library/react';
import React from 'react';
import Breadcrumb from '..';
import breadcrumb from '../__fixtures__';

describe('breadcrumb component', () => {
  it('should render correctly', () => {
    const { container } = render(<Breadcrumb {...breadcrumb} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
