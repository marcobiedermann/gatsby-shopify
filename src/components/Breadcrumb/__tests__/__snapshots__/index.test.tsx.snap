// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumb component should render correctly 1`] = `
<ol
  class="breadcrumb"
>
  <li
    class="breadcrumb__item"
  >
    <a
      href="/"
    >
      Home
    </a>
  </li>
</ol>
`;
